atlp-rwanda/e-comm-team-brogrammers-bn
# Feature(Authorization):Protecting routes #184581333

#### What does this PR do?

This PR is for feature of protecting routes so that only authorized users can access it

#### Description of Task to be completed?

this task is  to be done in following way

1. create a middleware  that will be imported whenever we have to protect a route
2. inside a that file there will be a function that will first check if there is token in bearer
3. after confirming that no token is there user will get a message to login or signup
4. and the if there is a token , function will check if token is valid and return an error otherwise allow user to access route


#### How should this be manually tested?

- after cloning the  the repo
-  type `npm install` to install all packages
- after  go to postman or where ever  you test your api and test a protected route like getting all users
- try different approaches like removing token and using invalid token to make sure that the api works

#### Any background context you want to provide?

N/a

#### What are the relevant pivotal tracker/Trello stories?

https://www.pivotaltracker.com/story/show/184581333

#### Screenshots (if appropriate)


#### Questions:
